ROADMAP.md

```diff
@@ -137,58 +238,73 @@ n/a
 
 > Summary: Currently our records are simple protobufs that get stored in the datastore. With this milestone, any record (provider or IPNS) will be a full IPLD object that follows the IPRS spec.
 
-- Tasks
-  - [ ] Technical Discussion: Decide if whether we should support two types of records or simply create another universe of the DHT
-  - [ ] Migrate go-ipfs to use IPRS
-  - [ ] js-libp2p-dht (also known as dht legacy)
-- Dependencies
-  - n/a
-- Requirements by other projects
-  - n/a
+### Tasks
 
-### Notes
+- [ ] Technical Discussion: Decide if whether we should support two types of records or simply create another universe of the DHT
+- [ ] Migrate go-ipfs to use IPRS
+- [ ] js-libp2p-dht (also known as dht legacy)
 
-n/a
+### Notes `NA`
 
 ### Expected date of completion: `Q4`
 
 ----------------------------------------------------------------
 
 ## Milestone 7 - Break DHT into Peer Routing and Content Routing
 
-> Summary: n/a
+> Summary: `NA`
 
-- Tasks
-  - [ ] Technical Discussion: Revisit proposed spec
-- Dependencies
-  - n/a
-- Requirements by other projects
-  - n/a
+### Tasks
+
+- [ ] Technical Discussion: Revisit proposed spec
 
 ### Notes
 
 This milestone is still very green, however something we know for sure we want to do.
 
-### Expected date of completion: `n/a`
+### Expected date of completion: `NA`
 
 ----------------------------------------------------------------
 
 ## Milestone 8 - Packet Switching / Overlay Network
 
 > Summary: The packet switched overlay network
 
-- Tasks
-  - [ ] Packet switch implemented and integrated into go-ipfs
-    - https://github.com/ipfs/notes/issues/143
-- Dependencies
-  - [ ] Multigram specced and implemented in Golang
-    - https://github.com/multiformats/multigram
-- Requirements by other projects
-  - n/a
+### Tasks
+
+- [ ] Packet switch implemented and integrated into go-ipfs
+  - https://github.com/ipfs/notes/issues/143
+
+### Dependencies
+
+- [ ] Multigram specced and implemented in Golang
+  - https://github.com/multiformats/multigram
 
 ### Notes
 
 The overlay network will not have any multi-hop routing for now.
 We only introduce the packet switch and continue being a direct single-hop network.
 
 ### Expected date of completion: `Q3`
+
+## Milestone 2 - Improve Connectivity of go-libp2p/go-ipfs
+
+> Summary: Improve the connectivity of go-libp2p. Have the technical discussions that will enable us to follow a plan to achieve a better connected graph.
+
+### Tasks
+
+- [ ] Technical Discussion: Define a strategy for connection closing
+  - https://github.com/libp2p/go-libp2p/issues/45
+  - https://github.com/ipfs/go-ipfs/issues/3065
+  - https://github.com/ipfs/notes/issues/110
+- [ ] Implement the strategy for connection closing in go-libp2p
+- [ ] Technical Discussion: Definte a strategy for Relay (or multiple Relay)
+  - https://github.com/ipfs/notes/issues/110
+- [ ] Implement Relay in go-libp2p
```

Comment:
Better question, is this doc just milestones in general? or is there some sort of weighting for Q4?